datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -154,11 +154,13 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
   /**
    * Move a partition for a datastream group according to the targetAssignment. As we are only allowed to mutate the
    * task once. It follow the steps
-   * Step 1) get the partitions that to be moved, and find out their source task
-   * Step 2) If the instance is the one we want to move, we find out the task which we should assign the partition
+   * Step 1) Process the targetAssignment to remove any partitions with no-op moves (partition currently assigned
+   *         to the same instance where the partition is to be moved)
+   * Step 2) Get the partitions that are to be moved, and find their source tasks
+   * Step 3) If the instance is the one we want to move, we choose a task to which we should assign the partition
```

Comment:
Also, can you please add the steps in the comments in the code, so it is clear which code is which step?

Reply:
done